sgv2-service-common/src/main/java/io/stargate/sgv2/common/grpc/DefaultStargateBridgeClient.java

```diff
@@ -76,120 +82,172 @@
   }
 
   @Override
-  public Response executeQuery(Query request) {
-    Response response =
-        ClientCalls.blockingUnaryCall(
-            channel, StargateBridgeGrpc.getExecuteQueryMethod(), callOptions, request);
-    if (response.hasSchemaChange()) {
-      // Ensure we're not holding a stale copy of the impacted keyspace by the time we return
-      // control to the caller
-      String keyspaceName = addTenantPrefix(response.getSchemaChange().getKeyspace());
-      schema.removeKeyspace(keyspaceName);
-    }
-    return response;
+  public CompletionStage<Response> executeQueryAsync(Query query) {
+    ClientCall<Query, Response> call =
+        channel.newCall(StargateBridgeGrpc.getExecuteQueryMethod(), callOptions);
+    UnaryStreamObserver<Response> observer = new UnaryStreamObserver<>();
+    ClientCalls.asyncUnaryCall(call, query, observer);
+    return observer
+        .asFuture()
+        .thenApply(
+            response -> {
+              if (response.hasSchemaChange()) {
+                // Ensure we're not holding a stale copy of the impacted keyspace by the time we
+                // return control to the caller
+                String keyspaceName = addTenantPrefix(response.getSchemaChange().getKeyspace());
+                schema.removeKeyspace(keyspaceName);
+              }
+              return response;
+            });
   }
 
   @Override
-  public Response executeBatch(Batch request) {
-    return ClientCalls.blockingUnaryCall(
-        channel, StargateBridgeGrpc.getExecuteBatchMethod(), callOptions, request);
+  public CompletionStage<Response> executeBatchAsync(Batch batch) {
+    ClientCall<Batch, Response> call =
+        channel.newCall(StargateBridgeGrpc.getExecuteBatchMethod(), callOptions);
+    UnaryStreamObserver<Response> observer = new UnaryStreamObserver<>();
+    ClientCalls.asyncUnaryCall(call, batch, observer);
+    return observer.asFuture();
   }
 
   @Override
-  public Optional<CqlKeyspaceDescribe> getKeyspace(String keyspaceName)
-      throws UnauthorizedKeyspaceException {
-    if (!authorizeSchemaRead(SchemaReads.keyspace(keyspaceName, sourceApi))) {
-      throw new UnauthorizedKeyspaceException(keyspaceName);
-    }
-    return getAuthorizedKeyspace(keyspaceName);
+  public CompletionStage<Optional<CqlKeyspaceDescribe>> getKeyspaceAsync(String keyspaceName) {
+    return authorizeSchemaReadAsync(SchemaReads.keyspace(keyspaceName, sourceApi))
+        .thenApply(
+            authorized -> {
+              if (!authorized) {
+                throw new UnauthorizedKeyspaceException(keyspaceName);
+              }
+              return authorized;
+            })
+        .thenCompose(__ -> getAuthorizedKeyspace(keyspaceName));
```

Comment:
this can be shorthanded?

```
        .thenCompose(
            authorized -> {
              if (!authorized) {
                throw new UnauthorizedKeyspaceException(keyspaceName);
              }
              return getAuthorizedKeyspace(keyspaceName);
            })